src/NHibernate/Util/TypeExtensions.cs

```diff
@@ -1,14 +1,19 @@
 using System;
 using System.Collections;
 using System.Collections.Generic;
+using System.Linq;
 
 namespace NHibernate.Util
 {
 	public static class TypeExtensions
 	{
 		public static bool IsEnumerableOfT(this System.Type type)
 		{
-			return type.IsGenericType && type.GetGenericTypeDefinition() == typeof(IEnumerable<>);
+			if (!type.IsGenericType) 
```

Comment:
Can you probably not change this method?

Reply:
Why? Let's obsolete it then? It's the only usage

Reply:
Yes, lets obsolete and inline it.